app/models/concerns/searchable.rb

```diff
@@ -0,0 +1,79 @@
+module Searchable
+  extend ActiveSupport::Concern
+
+  class_methods do
+    def search(query:, locale:)
+      where(id: select("DISTINCT #{name.underscore.pluralize}.id, search_type_results.rank").joins(sanitized_sql_for(locale, query)).pluck(:id))
+    end
+
+    def sanitized_sql_for(locale, query)
+      ActiveRecord::Base.sanitize_sql_array(
+        [
+          build_translation_search_sql,
+          { locale: locale.to_s, dictionary: dictionary_for(locale), query: query&.delete("'") }
+        ]
+      )
+    end
+
+    def dictionary_for(locale)
+      locale.to_s == 'en' ? 'english' : 'simple'
+    end
+
+    def build_translation_search_sql
+      table = name.underscore.pluralize
+      search_sql = <<-SQL.squish
+        INNER JOIN (
+          SELECT "#{table}"."id" AS search_id, (
+            ts_rank(
+              (
+                to_tsvector(:dictionary, coalesce(action_text_rich_texts_results.action_text_rich_texts_body::text, ''))
+                ||
+                to_tsvector(:dictionary, coalesce(mobility_string_translations_results.mobility_string_translations_value::text, ''))
+              ),
+              (
+                to_tsquery(:dictionary, ''' ' || :query || ' ''')
+              ), 0
+            )
+          ) AS rank FROM "#{table}"
+
+          LEFT OUTER JOIN (
+            SELECT "#{table}"."id" AS id, "action_text_rich_texts"."body"::text AS action_text_rich_texts_body
+            FROM "#{table}"
+            INNER JOIN "action_text_rich_texts" ON "action_text_rich_texts"."record_type" = '#{name}'
```

Comment:
In some cases we're passing in named parameter, e.g. `:locale` and `:dictionary` and in others using interpolated strings e.g. `name` and `table`. Is there a reason for that?